Libraries/PeriphDrivers/Include/MAX32655/i2s.h

```diff
@@ -91,11 +87,12 @@ typedef enum {
 typedef struct {
     mxc_i2s_ch_mode_t channelMode;
     mxc_i2s_stereo_t stereoMode;
-    mxc_i2s_wsize_t wordSize;
     mxc_i2s_justify_t justify;
     mxc_i2s_bitorder_t bitOrder;
     mxc_i2s_polarity_t wsPolarity;
-    mxc_i2s_samplesize_t sampleSize;
+    mxc_i2s_bitsword_t bitsWord;
+    mxc_i2s_wsize_t wordSize;
+    uint8_t sampleSize;
```

Comment:
There's the issue of preserving struct member names because current projects using I2S will be using the `mxc_i2s_samplesize_t sampleSize` member and the `mxc_i2s_samplesize_t` enum values instead of how it should be used for `uint8_t sampleSize` and `mxc_i2s_bitwords_t bitsWord`. This adds some complication... Name changes are always difficult to handle as we want to make things clear while also preventing any projects from breaking since these names have been released and in use for quite some time now.

We might need a discussion with @lorne-maxim and @Jake-Carter to figure out what the best approach is for this issue.

Reply:
Since `bitsWord` is completely new, I think `sampleSize` is the only issue (after https://github.com/Analog-Devices-MSDK/msdk/pull/847#discussion_r1445237701).

I think this would work:

```C
typedef enum {
    MXC_I2S_SAMPLESIZE_EIGHT = 8,
    MXC_I2S_SAMPLESIZE_SIXTEEN = 16,
    MXC_I2S_SAMPLESIZE_TWENTY = 20,
    MXC_I2S_SAMPLESIZE_TWENTYFOUR = 24,
    MXC_I2S_SAMPLESIZE_THIRTYTWO = 32
} mxc_i2s_samplesize_t __attribute__((deprecated)); // leave samplesize_t defined, but change enum values

typedef enum {
    MXC_I2S_BITSWORD_EIGHT,
    MXC_I2S_BITSWORD_SIXTEEN,
    MXC_I2S_BITSWORD_TWENTY,
    MXC_I2S_BITSWORD_TWENTYFOUR,
    MXC_I2S_BITSWORD_THIRTYTWO,
} mxc_i2s_bitsword_t;
```

So even if application code writes...

```C
req.bitsWord = MXC_I2S_BITSWORD_TWENTYFOUR;
req.sampleSize = MXC_I2S_SAMPLESIZE_EIGHT;
// ...
```
... sampleSize still functionally works as expected with the type change in the new struct definition.

The code actually builds cleanly without any warnings at all.  The only way you can get the deprecation to print is if you write this, which no one will actualy do...  so I don't know if the deprecation attribute is even worth it.

```C
mxc_i2s_samplesize_t test = MXC_I2S_SAMPLESIZE_EIGHT;
```

Reply:
Sounds applicable to me. I prefer not to include deprecated stuff in the API. IMHO 'crash and build it clean' or 'do not change at all' would be a good practice for the APIs. Surely, it depends on the popularity of the API to take this radical decision.

Reply:
I will resolve this thread in favor of https://github.com/Analog-Devices-MSDK/msdk/pull/847#discussion_r1452416765